test/e2e/main_test.go

```diff
@@ -161,14 +156,21 @@ func TestManualMeteringInstall(t *testing.T) {
 	}
 
 	for _, testCase := range testInstallConfigs {
-		t := t
 		testCase := testCase
 
 		if testCase.Skip {
 			continue
 		}
 
 		t.Run(testCase.Name, func(t *testing.T) {
+			t.Parallel()
+			var err error
+			var df *deployframework.DeployFramework
+
+			if df, err = deployframework.New(logger, runTestsLocal, runDevSetup, namespacePrefix, repoPath, repoVersion,
+				kubeConfig); err != nil {
+				logger.Fatalf("Failed to create a new deploy framework: %v", err)
+			}
```

Comment:
Is there a reason you needed to do this? We should be aiming towards parallelizing each deployment context instead of instantiating a new deploy framework per test case.

Reply:
Well it seemed like the easiest way to get separation between the namespaces. Otherwise the tests fail with context errors on the api calls: mismatched namespace to call.

Reply:
So I think we may want to re-think that approach as the `deployframework` structure is intended to encapsulate all metering installations. We may be able to get away with adding `t.Parallel()` to `TestManualMeteringInstall`, and again at the top of the helper function that top-level test calls, and go from there. I imagine the actual deployment of metering shouldn't be a problem, at least at the surface, but I imagine running the teardown method + cleanup script + post-install tests would need further refinement.